runtime/src/utils.rs

```diff
@@ -1,20 +1,20 @@
 /// Macro to set a value (e.g. when using the `parameter_types` macro) to either a production value
-/// or to an environment variable or testing value (in case the `fast-runtime` feature is selected).
+/// or to an environment variable or testing value (in case the `test-runtime` or `dev-runtime` feature is selected).
 ///
 /// Usage:
 /// ```Rust
 /// parameter_types! {
-///     pub LaunchPeriod: BlockNumber = conf!(mainnet: 7 * DAYS, testnet: 1);
-///     pub const VotingPeriod: BlockNumber = conf!(mainnet: 7 * DAYS, testnet: 1 * MINUTES);
+///     pub LaunchPeriod: BlockNumber = conf!(mainnet: 7 * DAYS, testnet: 1 * DAYS, devnet: HOURS);
+///     pub const VotingPeriod: BlockNumber = conf!(mainnet: 7 * DAYS, testnet: 5 * MINUTES, devnet: 1 * MINUTES);
 /// }
 /// ```
 #[macro_export]
 macro_rules! conf {
-    (mainnet: $prod:expr, testnet: $test:expr) => {
-        if cfg!(feature = "fast-runtime") {
-            $test
-        } else {
-            $prod
+    (mainnet: $prod:expr, testnet: $test:expr, devnet: $dev:expr) => {
+        match () {
+            _ if cfg!(feature = "test-runtime") => $test,
+            _ if cfg!(feature = "dev-runtime") => $dev,
+            _ => $prod,
```

Comment:
Maybe we can make this macro panic for _otherwise match arm_ and then use default features section in toml with new `mainnet-runtime` feauture?

Reply:
We can define reduced arity if we can use _testnet_ as _devnet_. E.g. `conf!(mainnet: 1 * DAYS, _: 1 * HOURS)`.

Reply:
Yes, I thought about that. At one point, I even considered adding it, but I thought it might be unnecessary. However, after further consideration, especially with the possibility of adding a 'stagenet' environment, it would be better if we strictly define the environment we're building the runtime for.

Reply:
I see your point about reducing arity by reusing the testnet configuration for devnet. However, I believe this might be unnecessary because configurations tend to change over time. On the devnet, configurations are likely to change frequently. For this reason, I think it’s better to keep the strict set of parameters for each environment.